gbtis/Kinect.cs

```diff
@@ -127,6 +141,26 @@ public class Kinect {
         }
 
         /// <summary>
+        /// Calculates the average finger position over the last
+        /// POINT_SAMPLE_SIZE frames
+        /// </summary>
+        /// <param name="x">The latest x coordinate</param>
+        /// <param name="y">The latest y coordinate</param>
+        /// <returns>The average position</returns>
+        private Point getAverageFingerTipPosition(float x, float y) {

```

Comment:
Agreed, but you only want to removefirst if size > 30

Maybe something like this?

    avg -= (samples.Count >= N) ? samples.RemoveFirst() : samples.First  / N;